Sources/KlaviyoSwift/KlaviyoState.swift

```diff
@@ -0,0 +1,134 @@
+//
+//  KlaviyoState.swift
+//  
+//
+//  Created by Noah Durell on 12/1/22.
+//
+
+import Foundation
+
+struct KlaviyoState: Codable {
+    var apiKey: String?
+    var email: String?
+    var anonymousId: String?
+    var phoneNumber: String?
+    var externalId: String?
+    var pushToken: String?
+    var queue: [KlaviyoAPI.KlaviyoRequest]
+    var requestsInFlight: [KlaviyoAPI.KlaviyoRequest]
+    var initialized = false
+    var flushing = false
+}
+
+// MARK: Klaviyo state persistence
+
+private func klaviyoStateFile(apiKey: String) -> URL {
+    let fileName = "klaviyo-\(apiKey)-state.json"
+    let directory = environment.fileClient.libraryDirectory()
+    return directory.appendingPathComponent(fileName, isDirectory: false)
+}
+
+private func storeKlaviyoState(state: KlaviyoState, at file: URL) {
+    do {
+        try environment.fileClient.write(environment.analytics.encodeJSON(state), file)
+    } catch {
+        environment.logger.error("Unable to save klaviyo state.")
+    }
+}
+
+private func removeStateFile(at file: URL) {
+    do {
+        try environment.fileClient.removeItem(file.path)
+    } catch {
+        environment.logger.error("Unable to remove state file.")
+    }
+}
+
+func loadKlaviyoStateFromDisk(apiKey: String) -> KlaviyoState {
+    let fileName = klaviyoStateFile(apiKey: apiKey)
+    guard environment.fileClient.fileExists(fileName.path) else {
+        return migrateLegacyDataToKlaviyoState(with: apiKey, to: fileName)
```

Comment:
Doesn't this mean migration will be run on even a clean install that never had the old SDK? Perhaps that is just fine.

Reply:
Yah I probably could do a little more to detect if it's needed but it should be very quick and only run once the first time.

Reply:
As long as its safe to run it on a clean install i wouldn't worry, just hard to tell from only reading the code.

Reply:
Ok...fair point...I'll see if I can make it a bit more readable in a future PR.